### ISSUE TITLE
provide a way to silence "failed to read..." matron spam (in repl)

### ISSUE THREAD

**Issue by pq** [monome/norns#404]
![image](https://user-images.githubusercontent.com/67586/41066785-03d89084-6998-11e8-86f2-210ddd6905c7.png)

periodically, i see streams like this coming from matron.

https://github.com/monome/norns/blob/484caad6cdadcf7d364eff4873c54a6adbebcdf4/matron/src/hardware/battery.c#L62

is it adding any value?  if not, it'd be great to have a way to filter this out.

assuming these messages should keep going to `stderr`, it may be easiest (but not best?) to do this in maiden?  (cc @ngwese)

dunno.

**Comment by simonvanderveldt** [comment]
Ideally we'd fix the cause of these failures, so far not sure what's causing them though.
See https://github.com/monome/norns-image/issues/43

**Comment by pq** [comment]
yeah.  not great to just drop them on the floor.  on the other hand, it's a bit noisy in the repl and could make folks think there's something up with the device.

**Comment by tehn** [comment]
i'm fine to just mute these battery warnings for now and open a ticket to investigate.

the code should also handle the fail more gracefully, ie, don't corrupt the previous read.

**Comment by simonvanderveldt** [comment]
Where does the repl get these logline from? Does it show all messages sent to `stderr`?

**Comment by pq** [comment]
@simonvanderveldt: i believe so.  @ngwese can confirm?

**Comment by ngwese** [comment]
the repl prints whatever `ws-wrapper` sends. i don’t remember if it captures both stdout and stderr or just stdout (from matron).

**Comment by pq** [comment]
@ngwese : is there any way to know whether we're getting (or sending) `stderr`?  (i'm thinking it would be great to be able to render the output differently.)

**Comment by pq** [comment]
an interesting possible data point regarding the cause of the errors: i've only noticed the errors when my norns has been powered (specifically by my macbook).  i don't think i've seen the errors when self-powered...

**Comment by pq** [comment]
> an interesting possible data point regarding the cause of the errors: i've only noticed the errors when my norns has been powered (specifically by my macbook). i don't think i've seen the errors when self-powered...

seeing this correlation again.  also fairly certain that the battery is fully charged when it occurs FWIW.

**Comment by markwheeler** [comment]
Had this happen today whilst norns was running on mains power, no other connections.
